### PR TITLE
[TT-5429] Tyk Pump Ignore Fields 

### PR DESCRIPTION
**Pull request by mativm02**
## Description

We are allowing users to ignore analytics record fields on their pumps. They can set this with any JSON tag of the [analytic record struct](https://github.com/TykTechnologies/tyk-pump/blob/master/analytics/analytics.go#L30) to ignore that field.
## Related Issue

https://tyktech.atlassian.net/jira/software/c/projects/TT/boards/42?modal=detail&selectedIssue=TT-5429
## Motivation and Context

It allows users to ignore fields that they don't really need.
## How This Has Been Tested

Unit tests were created and it was also tested with Pump Demo Mode and Mongo Pump.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: analytics/analytics.go [hunk 1]
```diff
@@ -348,3 +349,23 @@ func (a *AnalyticsRecord) IsGraphRecord() bool {
 
 	return false
 }
+
+func (a *AnalyticsRecord) RemoveIgnoredFields(ignoreFields []string) {
```
Comment (tbuchaillot) [inline analytics/analytics.go:353]: @mativm02 I think using reflection here makes this change way more dangerous. Can you try to refactor it using `github.com/fatih/structs`? We already use that lib in this type of stuff (for example https://github.com/TykTechnologies/tyk-pump/blob/1f0bc24934cd6d09829f5c0f6d6fe2e243f6b7ac/analytics/aggregate.go#L160), and IMO it's much safer

**Review by tbuchaillot — CHANGES_REQUESTED** [review]
Added some comment

File: analytics/analytics.go (hunk 1 above)
Comment (tbuchaillot) [reply analytics/analytics.go:353]: Nope @mativm02 - we didn't have any problem related to that repo (yet - I don't want to jinx it 😅 )

Comment (mativm02) [reply analytics/analytics.go:353]: @tbuchaillot done! now it's implementing `structs` library

**Review by tbuchaillot — APPROVED** [review]

**Review by sredxny — APPROVED** [review]
lgtm

**Pull request merged**